provider/pkg/provider/helm_release.go

```diff
@@ -299,9 +299,15 @@ func decodeRelease(pm resource.PropertyMap, label string) (*Release, error) {
 					if err != nil {
 						return nil, err
 					}
-					if err = yaml.Unmarshal(b, &values); err != nil {
+					valuesMap := map[string]any{}
+					if err = yaml.Unmarshal(b, &valuesMap); err != nil {
 						return nil, err
 					}
+					mergedValues, err := mergeMaps(values, valuesMap, false)
```

Comment:
I notice we might still deviate from upstream's behavior here -- this is Helm's logic:

https://github.com/helm/helm/blob/14d0c13e9eefff5b4a1b511cf50643529692ec94/pkg/cli/values/options.go#L108-L125

But we're using [github.com/imdario/mergo.Merge](https://github.com/darccio/mergo/blob/v0.3.13/merge.go#L345-L367). I haven't looked at this closely enough to see if there are behavioral differenced -- it's quite a bit more complex so I imagine there must be?

I would feel a lot better if we followed upstream's `mergeMaps` logic directly. It's not public but I wonder if we should just vendor the logic to stay closer to upstream. Or maybe this is an edge case not worth worrying about?

Reply:
I would definitely advocate for using Helm itself as much as possible, copying as necessary.

Reply:
Thanks for your fast feedbacks on that PR!

Using the same function as Helm to merge the maps sounds look I good idea.
`mergo.Merge` can do a deep merge on any struct, so the code is indeed way more complex. In our case we only need to deep merge two `map[string]any` so using the same function as Helm should be enough.
I can replace the call to `mergo.Merge` [here](https://github.com/pulumi/pulumi-kubernetes/blob/master/provider/pkg/provider/helm_release.go#L1331) with a call to a copy of the Helm merge maps. If that sounds good to you. As you said it is hard to anticipate any behavior changes seeing how complect the `mergoM.Merge` implementation is.
Do you think your test would catch any changes in the behavior of `mergeMaps`?

Let me know how I should handle License things if I copy the function from Helm. The code they use seems to be standard way to deep merge two `map[string]any` that can be found elsewhere on the internet

Reply:
I pushed a change to use upstream's merge logic. Both projects are Apache 2.0 and they don't include a `NOTICE` so I think it's sufficient to preserve attribution.

I left the existing `mergeMaps` implementation alone because I'd like to keep the scope of this change as small as possible for now. I'm not sure how well our tests cover that behavior.

Reply:
PR looks good to me like that. I think everything is in your hand to have it merged now. Let me know if I can do anything more to help with that.
What will be the release plan for that? I think this is a breaking change that can impact current users of the provider.

Reply:
> I think this is a breaking change that can impact current users of the provider.

It's breaking in the sense that behavior will change, yes. In this case it's a bug that this field doesn't agree (and has never agreed) with helm's behavior despite claiming to. I'm comfortable fixing the behavior instead of forcing users to live with it until a much later date for v5 -- or introducing another awkward field `valueYamlFilesThisIsTheOneYouReallyWant`.

Correctness is more important since the current behavior can potentially cause cascading issues by unexpectedly dropping values. If someone has been impacted by this, they might not realize it (in which case they could be sitting on a time bomb) or they might have worked around it by providing a union of values (in which case the change in behavior is less impactful). 

In any case, previews will show a diff with any impacted values and users can choose to modify their values accordingly.